### PR TITLE
Added protected route issue #26

### PR DESCRIPTION
**Pull request by lavnimusha**
### What this PR does (required):
if login credentials do not match or wrong url is typed current page is redirected to login page.

### Screenshots / Videos (required):
TBA

### Any information needed to test this feature (required):
With invalid credentials try to login.

### Any issues with the current functionality (optional):
functionality should be tested

### CODE REVIEW

**Review by ugur-akin — COMMENTED** [review]
Good stuff!

File: client/src/components/ProtectedRoute.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React, { Component } from 'react';
+import { Route, Redirect } from 'react-router-dom';
+import { useAuth } from '../context/useAuthContext';
+
+const ProtectedRoute = ({ path, component: Component, ...rest }: { path: string; component: any }) => {
+  const loggedInUser = useAuth();
+  return (
+    <Route
```
Comment (rajivtitus) [inline client/src/components/ProtectedRoute.tsx:8]: Maybe pass in the `...rest` props from above in addition to the default props that <Route> provides normally? :)

File: client/src/components/ProtectedRoute.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      render={(props) => {
+        if (loggedInUser) {
+          <Component />;
+        } else {
+          return <Redirect to={{ pathname: path, state: { from: props.location } }} />;
```
Comment (rajivtitus) [inline client/src/components/ProtectedRoute.tsx:13]: Won't we need to redirect to the path '/login' here?

**Review by rajivtitus — COMMENTED** [review]
Just some minor comments, please have a look and address when possible :)

File: client/src/components/ProtectedRoute.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import React, { Component } from 'react';
+import { Route, Redirect } from 'react-router-dom';
+import { useAuth } from '../context/useAuthContext';
+
+const ProtectedRoute = ({ path, component: Component, ...rest }: { path: string; component: any }) => {
+  const loggedInUser = useAuth();
```
Comment (rajivtitus) [inline client/src/components/ProtectedRoute.tsx:6]: Also, consider destructuring this, as the useAuth hook returns an object and we only need access to a single piece of state (i.e loggedInUser in this case)

**Review by rajivtitus — APPROVED** [review]
LGTM! Please address pending comments. Thank you!

File: client/src/components/ProtectedRoute.tsx (hunk 2 above)
Comment (lavnimusha) [reply client/src/components/ProtectedRoute.tsx]: resolved

File: client/src/App.tsx [hunk 4]
```diff
@@ -18,11 +18,9 @@ function App(): JSX.Element {
           <AuthProvider>
             <SocketProvider>
               <Switch>
+                <ProtectedRoute path="/dashboard" component={Dashboard} />
```
[the remark this answers is not in the document]
Comment (lavnimusha) [reply client/src/App.tsx]: added exact param.

File: client/src/components/ProtectedRoute.tsx (hunk 3 above)
Comment (lavnimusha) [reply client/src/components/ProtectedRoute.tsx]: destructuring added for "loggedInUser".

File: client/src/components/ProtectedRoute.tsx (hunk 1 above)
Comment (lavnimusha) [reply client/src/components/ProtectedRoute.tsx:8]: resolved.

**Pull request merged**